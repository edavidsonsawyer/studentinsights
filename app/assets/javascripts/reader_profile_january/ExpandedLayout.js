import React from 'react';
import PropTypes from 'prop-types';

export default class ExpandedLayout extends React.Component {
  render() {
    const {
      titleText,
      studentFirstName,
      onClose,
      materialsEl,
      strategiesEl,
      dataEl
    } = this.props;

    return (
      <div style={styles.root}>
        <div style={styles.header}>
          <div style={styles.title}>{titleText}</div>
          <div onClick={onClose} style={styles.close}>Close</div>
        </div>
        <div style={styles.columns}>
          <div style={{...styles.column, ...styles.materials}}>
            <div>{materialsEl}</div>
          </div>
          <div style={{...styles.column, ...styles.strategies}}>
            <div style={styles.subtitle}>Instructional strategies</div>
            <div>{strategiesEl}</div>
          </div>
          <div style={{...styles.column, ...styles.data}}>
            <div style={styles.subtitle}>Data for {studentFirstName}</div>
            <div>{dataEl}.</div>
          </div>
        </div>
      </div>
    );
  }
}
ExpandedLayout.propTypes = {
  titleText: PropTypes.string.isRequired,
  studentFirstName: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  materialsEl: PropTypes.node.isRequired,
  strategiesEl: PropTypes.node.isRequired,
  dataEl: PropTypes.node.isRequired
};


const styles = {
  root: {
    padding: 10
  },
  header: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingBottom: 10
  },
  title: {
    fontWeight: 'bold',
    fontSize: 16
  },
  close: {
    color: '#999',
    cursor: 'pointer'
  },
  columns: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between'
  },
  column: {
    marginRight: 20
  },
  materials: {
    flex: 5
  },
  strategies: {
    flex: 3
  },
  data: {
    flex: 5
  },
  subtitle: {
    color: '#333',
    marginBottom: 10
  }
};
