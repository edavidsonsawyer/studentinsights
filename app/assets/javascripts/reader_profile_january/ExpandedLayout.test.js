import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import {withDefaultNowContext} from '../testing/NowContainer';
import PerDistrictContainer from '../components/PerDistrictContainer';
import ExpandedLayout from './ExpandedLayout';

function testProps(props = {}) {
  return {
    titleText: 'First sound fluency',
    studentFirstName: 'Leo',
    onClose: jest.fn(),
    materialsEl: <div>materials!</div>,
    strategiesEl: <div>strategies!</div>,
    dataEl: <div>data!</div>
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(withDefaultNowContext(
    <PerDistrictContainer districtKey="somerville">
      <ExpandedLayout {...testProps()} />
    </PerDistrictContainer>
  ), el);
});


it('snapshots view', () => {
  const tree = renderer
    .create(
      <PerDistrictContainer districtKey="somerville">
        <ExpandedLayout {...testProps()} />
      </PerDistrictContainer>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});