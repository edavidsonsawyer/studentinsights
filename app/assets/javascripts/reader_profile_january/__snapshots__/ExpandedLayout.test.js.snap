// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  className="ExpandedLayout"
  style={
    Object {
      "padding": 10,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      First sound fluency
    </div>
    <div
      onClick={[MockFunction]}
      style={
        Object {
          "color": "#999",
          "cursor": "pointer",
        }
      }
    >
      Close
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 5,
          "marginRight": 20,
        }
      }
    >
      <div>
        <div>
          materials!
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 3,
          "marginRight": 20,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "color": "#333",
            "marginBottom": 10,
          }
        }
      >
        Instructional strategies
      </div>
      <div>
        <div>
          strategies!
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 5,
          "marginRight": 20,
        }
      }
    >
      <div
        style={
          Object {
            "color": "#333",
            "marginBottom": 10,
          }
        }
      >
        Somerville benchmarks for 
        Leo
      </div>
      <div>
        <div>
          data!
        </div>
      </div>
    </div>
  </div>
</div>
`;
